modules/internal/ChapelLocale.chpl

```diff
@@ -558,12 +558,12 @@ module ChapelLocale {
     // LocaleSpace -- an array of locales and its corresponding domain
     // which are used as the default set of targetLocales in many
     // distributions.
-    proc getDefaultLocaleSpace() const ref {
+    proc getDefaultLocaleSpace() const ref : chpl_emptyLocaleSpace.type {
       HaltWrappers.pureVirtualMethodHalt();
       return chpl_emptyLocaleSpace;
     }
 
-    proc getDefaultLocaleArray() const ref {
+    proc getDefaultLocaleArray() const ref : chpl_emptyLocales.type {
       HaltWrappers.pureVirtualMethodHalt();
       return chpl_emptyLocales;
```

Comment:
Is this stylistic or do we need it due to the `resolveSpecifiedReturnType` line added by this PR?

Reply:
Without it, I get e.g.

```
$CHPL_HOME/modules/internal/ChapelLocale.chpl:566: error: conflicting return type specified for 'AbstractRootLocale.getDefaultLocaleArray() [101921]: void'
$CHPL_HOME/modules/internal/localeModels/flat/LocaleModel.chpl:199: error:   overridden by 'RootLocale.getDefaultLocaleArray() [100368]: _ref(_array(unmanaged DefaultRectangularArr(1,int(64),one,locale,int(64))))'
```

(I think this might have to do with relatively recent changes Vass made to tread code after a `halt` as dead code).

The `resolveSpecifiedReturnType` line allows the code to override the compiler determining that the return type is `void` (because there are no non-dead returns).